src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_doing_flr_with_native_transactions.cs

```diff
@@ -14,9 +15,16 @@ public void Should_do_5_retries_by_default_with_native_transactions()
             Scenario.Define(() => new Context { Id = Guid.NewGuid() })
                     .WithEndpoint<RetryEndpoint>(b => b.Given((bus, context) => bus.SendLocal(new MessageToBeRetried { Id = context.Id })))
                     .AllowExceptions()
-                    .Done(c => c.ForwardedToErrorQueue || c.NumberOfTimesInvoked > 5)
+                    .Done(c => c.ForwardedToErrorQueue)
                     .Repeat(r => r.For(Transports.Default))
-                    .Should(c => Assert.AreEqual(5+1, c.NumberOfTimesInvoked, "The FLR should by default retry 5 times"))
+                    .Should(c =>
```

Comment:
Correct

On Wed, Sep 2, 2015 at 2:10 PM, Szymon Pobiega <notifications@github.com>
wrote:

> In
> src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_doing_flr_with_native_transactions.cs
> <https://github.com/Particular/NServiceBus/pull/2830#discussion_r38524355>
> :
>
> >                      .Repeat(r => r.For(Transports.Default))
> > -                    .Should(c => Assert.AreEqual(5+1, c.NumberOfTimesInvoked, "The FLR should by default retry 5 times"))
> > +                    .Should(c =>
>
> I think we decided at some point to try to avoid Should and Repeat and
> try to write tests in such a way so that we can just add asserts to the
> test method. Do you remember why @danielmarbach
> <https://github.com/danielmarbach> @andreasohlund
> <https://github.com/andreasohlund> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2830/files#r38524355>.
>